toolz/utils.py

```diff
@@ -6,4 +6,11 @@ def raises(err, lamda):
         return True
 
 
-no_default = '__no__default__'
+@object.__new__
+class no_default(object):
+    def __new__(self):
+        raise TypeError("cannot create 'no_default' instances")
+
+    def __str__(self):
+        return '<object no_default>'
+    __repr__ = __str__
```

Comment:
This seems like a timebomb though. What happens when you are processing some massive text file and this just happens to be in there? If this happens to someone in the wild, they will just feel frustrated and probably add some workaround instead of waiting for the library maintainers to fix it. I would argue that it is safer to defend against that.